// discord-interactions/dist/components.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TextStyleTypes = exports.ButtonStyleTypes = exports.MessageComponentTypes = void 0;
/**
 * The type of component
 * @see {@link https://discord.com/developers/docs/interactions/message-components#component-object-component-types}
 */
var MessageComponentTypes;
(function (MessageComponentTypes) {
    MessageComponentTypes[MessageComponentTypes["ACTION_ROW"] = 1] = "ACTION_ROW";
    MessageComponentTypes[MessageComponentTypes["BUTTON"] = 2] = "BUTTON";
    MessageComponentTypes[MessageComponentTypes["STRING_SELECT"] = 3] = "STRING_SELECT";
    MessageComponentTypes[MessageComponentTypes["INPUT_TEXT"] = 4] = "INPUT_TEXT";
})(MessageComponentTypes || (MessageComponentTypes = {}));
exports.MessageComponentTypes = MessageComponentTypes;
var ButtonStyleTypes;
(function (ButtonStyleTypes) {
    ButtonStyleTypes[ButtonStyleTypes["PRIMARY"] = 1] = "PRIMARY";
    ButtonStyleTypes[ButtonStyleTypes["SECONDARY"] = 2] = "SECONDARY";
    ButtonStyleTypes[ButtonStyleTypes["SUCCESS"] = 3] = "SUCCESS";
    ButtonStyleTypes[ButtonStyleTypes["DANGER"] = 4] = "DANGER";
    ButtonStyleTypes[ButtonStyleTypes["LINK"] = 5] = "LINK";
})(ButtonStyleTypes || (ButtonStyleTypes = {}));
exports.ButtonStyleTypes = ButtonStyleTypes;
var TextStyleTypes;
(function (TextStyleTypes) {
    TextStyleTypes[TextStyleTypes["SHORT"] = 1] = "SHORT";
    TextStyleTypes[TextStyleTypes["PARAGRAPH"] = 2] = "PARAGRAPH";
})(TextStyleTypes || (TextStyleTypes = {}));
exports.TextStyleTypes = TextStyleTypes;
